stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/mention/DefaultUserLookupHandler.kt

```diff
@@ -0,0 +1,93 @@
+package io.getstream.chat.android.ui.message.input.mention
+
+import com.ibm.icu.text.Transliterator
+import io.getstream.chat.android.client.logger.ChatLogger
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.android.ui.message.input.MessageInputView
+import java.text.Normalizer
+import kotlin.math.min
+
+private const val MAX_DISTANCE = 3
+private val regexUnaccent = "\\p{InCombiningDiacriticalMarks}+".toRegex()
+
+/*
+* Default implementation for MessageInputView.UserLookupHandler. This class ignores diacritics and upper case.
+* Tt uses levenshtein approximation so typos are included in the search. It is possible to choose a transliteration
+* in the class to conversions between languages are possible. It uses https://unicode-org.github.io/icu/userguide/icu4j/
+* for transliteration
+*/
+public class DefaultUserLookupHandler(
+    public var users: List<User>,
+    transliterationId: String? = null,
+) : MessageInputView.UserLookupHandler {
+
+    private var transliterator: Transliterator? = null
+    private val logger = ChatLogger.get("DefaultUserLookupHandler")
+
+    init {
+        transliterationId?.let(::setTransliterator)
+    }
+
+    private fun setTransliterator(id: String) {
+        if (Transliterator.getAvailableIDs().asSequence().contains(id)) {
+            this.transliterator = Transliterator.getInstance(id)
+        } else {
+            logger.logD("The id: $id for transliteration is not available")
+        }
+    }
+
+    override suspend fun handleUserLookup(query: String): List<User> {
```

Comment:
Question -> Do we also sort the list further based on l-distance or something else?

Reply:
No we do 👍